nexus/db-model/src/saga_types.rs

```diff
@@ -184,26 +184,35 @@ impl FromSql<SagaCachedStateEnum, Pg> for SagaCachedState {
 pub struct Saga {
     pub id: SagaId,
     pub creator: SecId,
-    pub template_name: String,
     pub time_created: chrono::DateTime<chrono::Utc>,
-    pub saga_params: serde_json::Value,
+    pub name: String,
+    pub saga_dag: serde_json::Value,
     pub saga_state: SagaCachedState,
     pub current_sec: Option<SecId>,
     pub adopt_generation: super::Generation,
     pub adopt_time: chrono::DateTime<chrono::Utc>,
 }
 
 impl Saga {
-    pub fn new(id: SecId, params: steno::SagaCreateParams) -> Self {
+    pub fn new(creator: SecId, params: steno::SagaCreateParams) -> Self {
         let now = chrono::Utc::now();
+
+        // This match will help us identify a case where Steno adds a new field
+        // to `SagaCreateParams` that we aren't persisting in the database.  (If
+        // you're getting a compilation failure here, you need to figure out
+        // what to do with the new field.  The assumption as of this writing is
+        // that we must store it into the database or we won't be able to
+        // properly recover the saga.)
+        let steno::SagaCreateParams { id, name, dag, state } = params;
```

Comment:
Nifty little safety feature :)

Reply:
I think I picked this up from @cbiffle.  I like it.